logic/transfer.js

```diff
@@ -153,29 +174,96 @@ Transfer.prototype.undoUnconfirmed = function (trs, sender, cb) {
 	return setImmediate(cb);
 };
 
+
 /**
- * Deletes blockId from transaction 
+ * @typedef {Object} transfer 
+ * @property {String} data
+ */
+Transfer.prototype.schema = {
+	id: 'transfer',
+	type: 'object',
+	properties: {
+		data: {
+			type: 'string',
+			minLength: 1,
+			maxLength: 64
+		}
+	}
+};
+
+/**
+ * Deletes blockId from transaction, and validates schema if asset exists.
  * @param {transaction} trs
  * @return {transaction}
  */
 Transfer.prototype.objectNormalize = function (trs) {
 	delete trs.blockId;
+
+	if (!trs.asset) {
+		return trs;
+	}
+
+	if (trs.asset.data === null || typeof trs.asset.data === 'undefined') {
+		delete trs.asset.data;
+	}
+
+	var report = library.schema.validate(trs.asset, Transfer.prototype.schema);
+
+	if (!report) {
+		throw 'Failed to validate transfer schema: ' + library.schema.getLastErrors().map(function (err) {
+			return err.message;
+		}).join(', ');
+	}
+
 	return trs;
 };
 
+Transfer.prototype.dbTable = 'transfer';
+
+Transfer.prototype.dbFields = [
+	'data',
+	'transactionId'
+];
+
+
 /**
+ * Checks if asset exists, if so, returns value, otherwise returns null.
  * @param {Object} raw
  * @return {null}
  */
 Transfer.prototype.dbRead = function (raw) {
+	if (raw.tf_data) {
+		return { data: raw.tf_data };
+	}
+
 	return null;
 };
 
 /**
+ * Checks if asset exists, if so, returns transfer table promise, otherwise returns null.
  * @param {transaction} trs
- * @return {null}
+ * @return {Object} {table:transfer, values: data and transactionId} or null.
```

Comment:
Should I add @class or should I use @typedef? 
http://usejsdoc.org/tags-typedef.html